tests/app/forms/validation/test_date_check_validator.py

```diff
@@ -1,198 +1,90 @@
-
-    def test_date_type_validator_missing_day(self):
-        validator = DateCheck()
-
-        mock_form = Mock()
-        mock_form.data = "2016-12-"
-        mock_form.year.data = "2016"
-
-        mock_field = Mock()
-
-        with self.assertRaises(StopValidation) as ite:
-            validator(mock_form, mock_field)
-
-        self.assertEqual(error_messages["INVALID_DATE"], str(ite.exception))
-
-    def test_date_type_validator_missing_month(self):
-        validator = DateCheck()
-
-        mock_form = Mock()
-        mock_form.data = "2016--03"
-        mock_form.year.data = "2016"
-
-        mock_field = Mock()
-
-        with self.assertRaises(StopValidation) as ite:
-            validator(mock_form, mock_field)
-
-        self.assertEqual(error_messages["INVALID_DATE"], str(ite.exception))
-
-    def test_date_type_validator_missing_year(self):
-        validator = DateCheck()
-
-        mock_form = Mock()
-        mock_form.data = "-12-03"
-
-        mock_field = Mock()
-
-        with self.assertRaises(StopValidation) as ite:
-            validator(mock_form, mock_field)
-
-        self.assertEqual(error_messages["INVALID_DATE"], str(ite.exception))
-
-    def test_date_type_validator_invalid_day(self):
-        validator = DateCheck()
-
-        mock_form = Mock()
-        mock_form.data = "2016-12-40"
-        mock_form.year.data = "2016"
-
-        mock_field = Mock()
-
-        with self.assertRaises(StopValidation) as ite:
-            validator(mock_form, mock_field)
-
-        self.assertEqual(error_messages["INVALID_DATE"], str(ite.exception))
-
-    def test_date_type_validator_invalid_month(self):
-        validator = DateCheck()
-
-        mock_form = Mock()
-        mock_form.data = "2016-13-20"
-        mock_form.year.data = "2016"
-
-        mock_field = Mock()
-
-        with self.assertRaises(StopValidation) as ite:
-            validator(mock_form, mock_field)
-
-        self.assertEqual(error_messages["INVALID_DATE"], str(ite.exception))
-
-    def test_date_type_validator_invalid_year(self):
-        validator = DateCheck()
-
-        mock_form = Mock()
-        mock_form.data = "20000-12-20"
-        mock_form.year.data = "20000"
-
-        mock_field = Mock()
-
-        with self.assertRaises(StopValidation) as ite:
-            validator(mock_form, mock_field)
-
-        self.assertEqual(error_messages["INVALID_DATE"], str(ite.exception))
-
-    def test_date_type_validator_invalid_year_digits(self):
-        validator = DateCheck()
-
-        mock_form = Mock()
-        mock_form.data = "2-12-20"
-        mock_form.year.data = "2"
-
-        mock_field = Mock()
-
-        with self.assertRaises(StopValidation) as ite:
-            validator(mock_form, mock_field)
-
-        self.assertEqual(error_messages["INVALID_DATE"], str(ite.exception))
-
-    def test_date_type_validator_invalid_month_digits(self):
-        validator = DateCheck()
-
-        mock_form = Mock()
-        mock_form.data = "2020--2-20"
```

Comment:
I couldn't see the value of the 3 tests with invalid_digits, (i.e 2020--2-20) Naturally we could be here all day with different combinations (letters, characters, too long etc) So I just concentrated on missing parts and invalid dates, i.e a month with 13, a day with 32 , an incorrect leap year

Reply:
Looks much better now!